ai21_tokenizer/jamba_instruct_tokenizer.py

```diff
@@ -97,22 +97,34 @@ async def create(
     async def encode(self, text: str, **kwargs) -> List[int]:
         if not self._tokenizer:
             await self._init_tokenizer()
-        return await asyncio.to_thread(self._encode, text, **kwargs)
+        return await asyncio.get_running_loop().run_in_executor(
+            executor=None, func=lambda: self._encode(text=text, **kwargs)
+        )
 
     async def decode(self, token_ids: List[int], **kwargs) -> str:
         if not self._tokenizer:
             await self._init_tokenizer()
-        return await asyncio.to_thread(self._decode, token_ids, **kwargs)
+        return await asyncio.get_running_loop().run_in_executor(
+            executor=None, func=lambda: self._decode(token_ids=token_ids, **kwargs)
+        )
 
     async def convert_tokens_to_ids(self, tokens: Union[str, List[str]]) -> Union[int, List[int]]:
         if not self._tokenizer:
             await self._init_tokenizer()
-        return await asyncio.to_thread(self._convert_tokens_to_ids, tokens)
+        return await asyncio.get_running_loop().run_in_executor(
+            None,
+            self._convert_tokens_to_ids,
```

Comment:
These don't need `lambda`?

In general, I think it would be best to create a protected method which will be the one to make the `asyncio.get_running_loop` calls so it will be in a single place like 


```python
async def _make_async_call(callback, **arguments):
    return await asyncio.get_running_loop.run_in_executor(...)
```